### PR TITLE
blog: announce grpc-zero

### PR DESCRIPTION
**Pull request by andreaTP**
Blog post announcing gRPC Zero availability as experimental.

### CODE REVIEW

File: _posts/2025-09-05-grpc-zero.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+-------------------
+
+= Quarkus gRPC Zero
+
+*Make gRPC code generation portable: no native protoc, no surprises.*
+
+== TL;DR
+
+Quarkus gRPC Zero brings gRPC code generation into the JVM so you no longer need native 'protoc' binaries. Add the extension, build your project, and the generated stubs appear just like before.
+The important outcome is consistent, portable builds across developer machines, CI, containers, and even unusual architectures.
```
Comment (jponge) [inline _posts/2025-09-05-grpc-zero.adoc:18]: I'd add a line break here so it stands up as a paragraph.

File: _posts/2025-09-05-grpc-zero.adoc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+== Why this matters
+
+If you have spent time wrestling with platform-specific 'protoc' binaries, cross-compiled plugins, or flaky CI images, you know the cost: slow onboarding, fragile builds, extra Docker layers, and ongoing maintenance. Quarkus gRPC Zero removes that operational burden so teams can focus on APIs and features instead of shipping native binaries for every target OS and architecture.
```
Comment (jponge) [inline _posts/2025-09-05-grpc-zero.adoc]: `protoc` (to have fixed-width font)

File: _posts/2025-09-05-grpc-zero.adoc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+== What Quarkus gRPC Zero does
+
+Quarkus gRPC Zero runs the `protoc` compilation inside the JVM as a pure Java codegen step.
+From a developer point of view nothing changes: you keep writing '.proto' files, run your Quarkus build, and use the generated sources.
```
Comment (jponge) [inline _posts/2025-09-05-grpc-zero.adoc]: Same, `.proto`

Comment (jponge) [reply _posts/2025-09-05-grpc-zero.adoc]: plus a  line break

File: _posts/2025-09-05-grpc-zero.adoc [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+The difference is that builds are portable and predictable on any JVM host.
+
+== Benefits you will see
```
Comment (jponge) [inline _posts/2025-09-05-grpc-zero.adoc]: Just _Benefits_

File: _posts/2025-09-05-grpc-zero.adoc [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Outcome: the developer ergonomics are unchanged, but there are no native tools to manage.
+
+== When to choose this extension
+
+* You want hermetic, reproducible builds with fewer external artifacts.
+* You support multiple platforms or unusual architectures and want to avoid maintaining binaries.
+
+== Current status and roadmap
+
+Quarkus gRPC Zero is ready for early adopters.
```
Comment (jponge) [inline _posts/2025-09-05-grpc-zero.adoc]: I'd put a line break after that sentence.

File: _posts/2025-09-05-grpc-zero.adoc [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+* Clone or add the extension to a test project.
+* Build on your laptop, in CI, and in any container or architecture you support.
+* If you see an error, open a ticket at the project repository: https://github.com/quarkiverse/quarkus-grpc-zero/issues[quarkiverse/quarkus-grpc-zero issues]
+
+*Call to action: try the extension today and report the results.* Your reports will directly shape the project and help us make code generation reliable for everyone.
+
+== Closing
+
+Quarkus gRPC Zero is about outcomes: consistent builds and no more native 'protoc' maintenance.
```
Comment (jponge) [inline _posts/2025-09-05-grpc-zero.adoc]: `protoc` :)

File: _posts/2025-09-05-grpc-zero.adoc (hunk 1 above)
Comment (andreaTP) [reply _posts/2025-09-05-grpc-zero.adoc:18]: done

File: _posts/2025-09-05-grpc-zero.adoc (hunk 2 above)
Comment (andreaTP) [reply _posts/2025-09-05-grpc-zero.adoc]: :+1:

File: _posts/2025-09-05-grpc-zero.adoc (hunk 3 above)
Comment (andreaTP) [reply _posts/2025-09-05-grpc-zero.adoc]: :+1:

File: _posts/2025-09-05-grpc-zero.adoc (hunk 4 above)
Comment (andreaTP) [reply _posts/2025-09-05-grpc-zero.adoc]: :+1:

**Comment by gsmet** [comment]
(I let you do it in case you don't want to publish on a Friday :))

**Comment by andreaTP** [comment]
Thanks @gsmet ! Just give me a date somewhere next week :slightly_smiling_face:

**Comment by andreaTP** [comment]
Tentatively updated the date to tomorrow.

File: _posts/2025-09-30-grpc-zero.adoc [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+== TL;DR
+
+Quarkus gRPC Zero brings gRPC code generation into the JVM so you no longer need native 'protoc' binaries. Add the extension, build your project, and the generated stubs appear just like before.
+
+The important outcome is consistent, portable builds across developer machines, CI, containers, and even unusual architectures.
+
+== Why this matters
+
+If you have spent time wrestling with platform-specific `protoc` binaries, cross-compiled plugins, or flaky CI images, you know the cost: slow onboarding, fragile builds, extra Docker layers, and ongoing maintenance. Quarkus gRPC Zero removes that operational burden so teams can focus on APIs and features instead of shipping native binaries for every target OS and architecture.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
If you have spent time wrestling with platform-specific `protoc` binaries, cross-compiled plugins, you know the cost: slow onboarding, fragile builds, extra Docker layers, and ongoing maintenance. Quarkus gRPC Zero removes that operational burden so teams can focus on APIs and features instead of trying to find the right combination of dependencies to compile `proto` files.
```

File: _posts/2025-09-30-grpc-zero.adoc [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+== What Quarkus gRPC Zero does
+
+Quarkus gRPC Zero runs the `protoc` compilation inside the JVM as a pure Java codegen step.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
Quarkus gRPC Zero runs the `protoc` compilation inside the JVM as a pure Quarkus _codegen_  module.
```

File: _posts/2025-09-30-grpc-zero.adoc [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+From a developer point of view nothing changes: you keep writing `.proto` files, run your Quarkus build, and use the generated sources.
+
+The difference is that builds are portable and predictable on any JVM host.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
The difference is that builds are portable and predictable on any JVM host. It avoids having to download dozen of dependencies to handle every OS/architecture combination.
```

File: _posts/2025-09-30-grpc-zero.adoc [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+== Benefits
+
+* Portable builds that behave the same on laptops, CI, containers, and edge devices.
+* Simpler CI and smaller images because you do not need to package native executables.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]: No that's not true, it does not change the size fo the artifact, only the amount of dependency you download.

File: _posts/2025-09-30-grpc-zero.adoc [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+* Less maintenance for platform teams who no longer manage platform-specific toolchains.
+* A small, self-contained Java dependency that performs code generation.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
* A small, self-contained Java dependency that performs `proto` file compilation.
```

File: _posts/2025-09-30-grpc-zero.adoc [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+  </exclusions>
+</dependency>
+<dependency>
+  <groupId>io.quarkiverse.grpc.zero</groupId>
+  <artifactId>quarkus-grpc-zero</artifactId>
+  <version>VERSION</version>
+</dependency>
+----
+
+If you are migrating from an existing Quarkus gRPC setup, you need to exclude the 'quarkus-grpc-codegen' artifact from your 'quarkus-grpc' dependency and add 'quarkus-grpc-zero' instead.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
If you are migrating from an existing Quarkus gRPC setup, you need to exclude the 'quarkus-grpc-codegen' artifact from your 'quarkus-grpc' dependency and add 'quarkus-grpc-zero' instead as a drop-in replacement.
```

File: _posts/2025-09-30-grpc-zero.adoc [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+Build your project as usual and generated sources will appear during the build step.
+
+== Example workflow
+
+. Add the 'quarkus-grpc' dependency with exclusions for 'quarkus-grpc-codegen' and include 'quarkus-grpc-zero'.
+. Keep authoring '.proto' files as before.
+. Run the Quarkus build. Generated stubs will be produced on the JVM and compilation completes normally.
+
+Outcome: the developer ergonomics are unchanged, but there are no native tools to manage.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
The developer ergonomics are unchanged, but there are no native tools invoked during the process.
```

File: _posts/2025-09-30-grpc-zero.adoc [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+== When to choose this extension
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]: I would remove that section.

File: _posts/2025-09-30-grpc-zero.adoc [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+* You want hermetic, reproducible builds with fewer external artifacts.
+* You support multiple platforms or unusual architectures and want to avoid maintaining binaries.
+
+== Current status and roadmap
+
+Quarkus gRPC Zero is ready for early adopters.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
Quarkus gRPC Zero is  currently experimental but ready for early adopters.
```

File: _posts/2025-09-30-grpc-zero.adoc (hunk 14 above)
Comment (cescoffier) [reply _posts/2025-09-30-grpc-zero.adoc]: Can we have a "Behind the curtain" section instead mentioning WASM, and Chicory?

File: _posts/2025-09-30-grpc-zero.adoc [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+It passes integration tests and works in typical Quarkus builds.
+We are actively improving the project and welcome feedback, real-world testing, and bug reports to guide stabilization and future features.
+
+== Try it and report any errors
+
+Please try Quarkus gRPC Zero in your projects.
+We want real-world feedback and we especially want to hear about any errors, edge cases, or surprising behavior you encounter.
+We are happy to quickly to turn things around to fix outstanding bugs.
+
+* Clone or add the extension to a test project.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]: I would only keep the last bullet point:

Report an issue on ....

File: _posts/2025-09-30-grpc-zero.adoc [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+* Build on your laptop, in CI, and in any container or architecture you support.
+* If you see an error, open a ticket at the project repository: https://github.com/quarkiverse/quarkus-grpc-zero/issues[quarkiverse/quarkus-grpc-zero issues]
+
+*Call to action: try the extension today and report the results.* Your reports will directly shape the project and help us make code generation reliable for everyone.
```
Comment (cescoffier) [inline _posts/2025-09-30-grpc-zero.adoc]:
```suggestion
Your reports will shape the project and help us make code generation reliable for everyone.
```

File: _posts/2025-09-30-grpc-zero.adoc (hunk 10 above)
Comment (andreaTP) [reply _posts/2025-09-30-grpc-zero.adoc]: fixed

File: _posts/2025-09-30-grpc-zero.adoc (hunk 14 above)
Comment (andreaTP) [reply _posts/2025-09-30-grpc-zero.adoc]: applied both suggestions

File: _posts/2025-09-30-grpc-zero.adoc (hunk 16 above)
Comment (andreaTP) [reply _posts/2025-09-30-grpc-zero.adoc]: done

File: _posts/2025-10-16-grpc-zero.adoc [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+
+Quarkus gRPC Zero is  currently experimental but ready for early adopters.
+
+It passes integration tests and works in typical Quarkus builds.
+We are actively improving the project and welcome feedback, real-world testing, and bug reports to guide stabilization and future features.
+
+== Under the Hood
+
+The extension embeds a version of `libprotobuf`, compiled to WebAssembly (with the CLI stripped out) and translated into pure Java bytecode thanks to https://chicory.dev[Chicory].
+The result is a self-contained JAR that provides the full `protoc` engine capabilities (including plugin support) and runs on any JVM, transparently and portably across platforms.
```
Comment (cescoffier) [inline _posts/2025-10-16-grpc-zero.adoc:83]: Perfect!

**Comment by andreaTP** [comment]
I was waiting a post on the socials to make a bit of noise about this, should I wait a little more or post it independently?

cc. @insectengine @gsmet
